core/ecschema-editing/src/Differencing/SchemaDifferenceValidator.ts

```diff
@@ -462,6 +438,28 @@ class SchemaDifferenceValidationVisitor implements ISchemaDifferenceVisitor {
   public async visitUnitSystemDifference(entry: UnitSystemDifference) {
     await this.visitSchemaItemDifference(entry, await this._targetSchema.getItem(entry.itemName));
   }
+
+  /**
+   * Recursive synchronous function to figure whether a given class derived from
+   * a class with the given baseClass name.
+   */
+  private async derivedFrom(ecClass: ECClass | undefined, baseClassKey: Readonly<SchemaItemKey>): Promise<boolean> {
+    if (ecClass === undefined) {
+      return false;
+    }
+
+    // First check for name which must be same in any case...
+    if (ecClass.name === baseClassKey.name) {
+      // ... then check if the class is in the same schema as the expected base class...
+      if(ecClass.schema.name === baseClassKey.schemaName)
+        return true;
+      // ... if not, whether it's in the other schema, which could be the case if the base class
+      // gets added to the target schema...
+      if(ecClass.schema.name === this._sourceSchema.name || ecClass.schema.name === this._targetSchema.name)
```

Comment:
I don't understand how the ecClass in this case could have the target schema name because you load the ecClass from the source.

I think the right check is that if the ecClass.schema.name === source AND baseClassKey.schemaName === target then we can consider them equal.


You should be able to make a test case where the current code fails when both classes have the same name but the target class is defined in a referenced schema and the source class is defined in the source schema.